src/Kentico.Xperience.Lucene.Core/Indexing/DefaultLuceneConfigurationStorageService.cs

```diff
@@ -263,6 +287,50 @@ public bool TryEditIndex(LuceneIndexModel configuration)
                     }
                 }
             }
+
+            var removeContentTypesQuery = contentTypeProvider
+                .Get()
+                .WhereIn(nameof(LuceneContentTypeItemInfo.LuceneContentTypeItemIncludedPathItemId), removedPaths.Select(x => x.LuceneIncludedPathItemId).ToArray());
+
+            contentTypeProvider.BulkDelete(removeContentTypesQuery);
+
+            var allExistingContentTypes = await contentTypeProvider
+                .Get()
+                .WhereIn(nameof(LuceneContentTypeItemInfo.LuceneContentTypeItemIncludedPathItemId), configuration.Paths.Select(x => int.Parse(x.Identifier ?? "0")).ToArray())
+                .GetEnumerableTypedResultAsync();
+
+            int[] removedContentTypeIdsFromEditedPaths = allExistingContentTypes
+                .Where(x => !configuration.Paths
+                    .Any(y => y.ContentTypes
+                        .Exists(z => x.LuceneContentTypeItemIncludedPathItemId.ToString() == y.Identifier && x.LuceneContentTypeItemContentTypeName == z.ContentTypeName))
+                )
+                .Select(x => x.LuceneContentTypeItemId)
+                .ToArray();
+
+            contentTypeProvider.BulkDelete(contentTypeProvider.Get().WhereIn(nameof(LuceneContentTypeItemInfo.LuceneContentTypeItemId), removedContentTypeIdsFromEditedPaths));
+
+            foreach (var path in existingPaths)
+            {
+                path.LuceneIncludedPathItemAliasPath = configuration.Paths.Single(x => x.Identifier == path.LuceneIncludedPathItemId.ToString()).AliasPath;
+                path.Update();
+
+                foreach (var contentType in configuration.Paths
+                    .Single(x => x.Identifier == path.LuceneIncludedPathItemId.ToString())
```

Comment:
This is not readable, please move content types filtering into separate well-named variable and use that variable in foreach instead.